models/DataObject/ClassDefinition/Data/CalculatedValue.php

```diff
@@ -140,7 +140,7 @@ public function setCalculatorExpression(?string $calculatorExpression): void
      */
     public function getDataForQueryResource(mixed $data, DataObject\Concrete $object = null, array $params = []): ?string
     {
-        return $data;
+        return (string) $data;
```

Comment:
Should we handle null values?
```suggestion
        if ($data !== null) {
            return (string) $data;   
        }
        return null;
```

Reply:
According to the return type: yes.